source/code/plugin/filter_cadvisor2mdm.rb

```diff
@@ -0,0 +1,235 @@
+            end
+            @log.debug "AKS_REGION #{aks_region}"
+            custom_metrics_regions_arr = @custom_metrics_azure_regions.split(',')
+            custom_metrics_regions_hash = custom_metrics_regions_arr.map {|x| [x.downcase,true]}.to_h
+
+            @log.debug "Custom Metrics Regions Hash #{custom_metrics_regions_hash}"
+
+            if custom_metrics_regions_hash.key?(aks_region.downcase)
+                @log.debug "Returning true for check_custom_metrics_availability"
+                true
+            else 
+                @log.debug "Returning false for check_custom_metrics_availability"
+                false
+            end
+        end
+
+		def shutdown
+			super
+		end
+
+        def filter(tag, time, record)
+            begin
+                if @process_incoming_stream
+                    object_name = record['DataItems'][0]['ObjectName']
+                    counter_name = record['DataItems'][0]['Collections'][0]['CounterName']
+                    if object_name == @@object_name_k8s_node && @metrics_to_collect_hash.key?(counter_name.downcase)
+                        percentage_metric_value = 0.0
+
+                        # Compute and send % CPU and Memory
+                        metric_value = record['DataItems'][0]['Collections'][0]['Value']
+                        if counter_name.downcase == @@cpu_usage_nano_cores
+                            metric_name = @@cpu_usage_milli_cores
+                            metric_value = metric_value/1000000
+                            if @cpu_capacity != 0.0
+                                percentage_metric_value = (metric_value*1000000)*100/@cpu_capacity
+                            end
+                        end
+
+                        if counter_name.start_with?("memory")
+                            metric_name = counter_name
+                            if @memory_capacity != 0.0
+                                percentage_metric_value = metric_value*100/@memory_capacity
+                            end
+                        end 
+                        return get_metric_records(record, metric_name, metric_value, percentage_metric_value)
+                    else 
+                        return []
+                    end
+                else
+                    return []
+                end
+            rescue Exception => e
+                @log.info "Error processing cadvisor record Exception: #{e.class} Message: #{e.message}"
+                ApplicationInsightsUtility.sendExceptionTelemetry(e.backtrace)
+                return []
+            end
+        end
+
+        def ensure_cpu_memory_capacity_set
+
+            @log.info "ensure_cpu_memory_capacity_set @cpu_capacity #{@cpu_capacity} @memory_capacity #{@memory_capacity}"
+            if @cpu_capacity != 0.0 && @memory_capacity != 0.0
```

Comment:
These 2 not class vars, i guess we are making a call everytime (as they will be 0) ??